lib/srv/sessiontracker.go

```diff
@@ -181,6 +182,33 @@ func (s *SessionTracker) WaitForStateUpdate(initialState types.SessionState) typ
 	}
 }
 
+// SubscribeToStateUpdate waits for the tracker's state to be updated and sends the new states over a channel
+func (s *SessionTracker) SubscribeToStateUpdate(ctx context.Context) <-chan types.SessionState {
+	ch := make(chan types.SessionState, 1)
+
+	go func() {
+		s.trackerCond.L.Lock()
+		defer s.trackerCond.L.Unlock()
```

Comment:
Do you need to hold the lock while you wait for the channel receive? This seems likely to deadlock, or at least cause contention.